### PR TITLE
Create/open file on WASI targets should have POLL_FD_READWRITE rights

### LINKED ISSUE #14324: std: std.fs.Dir.createFile on WASI targets should include POLL_FD_READWRITE right by default
### Zig Version

0.11.0-dev.1183+2b9478ce1

### Steps to Reproduce and Observed Behavior

You can see this plainly in the source:

https://github.com/ziglang/zig/blob/bb15e4057c9c8bb22084990de475ab10a44592c0/lib/std/fs.zig#L1273-L1286

The right `w.RIGHT.POLL_FD_READWRITE` is not included and there is no way to add it using this interface. This makes it so that the WASI `poll_oneoff` (basically POSIX `poll(2)`) will return EPERM. It makes more sense to me that a file created should implicitly be poll-able.

### Expected Behavior

The WASI spec says:

> poll_fd_readwrite: bool If [rights::fd_read](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#rights.fd_read) is set, includes the right to invoke [poll_oneoff](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#poll_oneoff) to subscribe to [eventtype::fd_read](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#eventtype.fd_read). If [rights::fd_write](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#rights.fd_write) is set, includes the right to invoke [poll_oneoff](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#poll_oneoff) to subscribe to [eventtype::fd_write](https://github.com/WebAssembly/WASI/blob/main/phases/snapshot/docs.md#eventtype.fd_write).

Therefore, above the rights already granted, there is no risk or downside to setting this to true. If the user doesn't request read access in CreateFlags, `poll_fd_readwrite` doesn't implicitly grant it -- both that AND `fd_read` must be true -- so there isn't any elevation of privs.

This is a really easy fix, I'll try to drop a PR if someone doesn't beat me.

### PR DESCRIPTION
**Pull request by achan1989**
Fixes #14324

No testing performed. Original issue says "you can see this plainly in the source" -- agreed.

### CODE REVIEW

File: lib/std/fs/Dir.zig [hunk 1]
```diff
@@ -809,6 +809,7 @@ pub fn openFile(self: Dir, sub_path: []const u8, flags: File.OpenFlags) File.Ope
             base.FD_TELL = true;
             base.FD_SEEK = true;
             base.FD_FILESTAT_GET = true;
+            base.POLL_FD_READWRITE = true;
```
Comment (squeek502) [inline lib/std/fs/Dir.zig:812]:
```suggestion
            // If rights::fd_read is set, includes the right to invoke poll_oneoff to subscribe to eventtype::fd_read
            base.POLL_FD_READWRITE = true;
```

File: lib/std/fs/Dir.zig [hunk 2]
```diff
@@ -821,6 +822,7 @@ pub fn openFile(self: Dir, sub_path: []const u8, flags: File.OpenFlags) File.Ope
             base.FD_ADVISE = true;
             base.FD_FILESTAT_SET_TIMES = true;
             base.FD_FILESTAT_SET_SIZE = true;
+            base.POLL_FD_READWRITE = true;
```
Comment (squeek502) [inline lib/std/fs/Dir.zig:825]:
```suggestion
            // If rights::fd_write is set, includes the right to invoke poll_oneoff to subscribe to eventtype::fd_write.
            base.POLL_FD_READWRITE = true;
```

File: lib/std/fs/Dir.zig [hunk 3]
```diff
@@ -982,6 +984,7 @@ pub fn createFile(self: Dir, sub_path: []const u8, flags: File.CreateFlags) File
                 .FD_FILESTAT_SET_TIMES = true,
                 .FD_FILESTAT_SET_SIZE = true,
                 .FD_FILESTAT_GET = true,
+                .POLL_FD_READWRITE = true,
```
Comment (squeek502) [inline lib/std/fs/Dir.zig:987]:
```suggestion
                // If rights::fd_read is set, includes the right to invoke poll_oneoff to subscribe to eventtype::fd_read.
                // If rights::fd_write is set, includes the right to invoke poll_oneoff to subscribe to eventtype::fd_write.
                .POLL_FD_READWRITE = true,
```

**Review by squeek502 — COMMENTED** [review]
I'm not familiar with this right/`poll_oneoff`, so I don't have an opinion on setting this right unconditionally.

It might be nice to include quotes from [the documentation](https://github.com/WebAssembly/WASI/blob/8777909de3d1cb5cdaee16f98d76c572d4b88b16/legacy/preview1/docs.md#rights.poll_fd_readwrite), though.

**Comment by achan1989** [comment]
> It might be nice to include comments with quotes from [the documentation](https://github.com/WebAssembly/WASI/blob/8777909de3d1cb5cdaee16f98d76c572d4b88b16/legacy/preview1/docs.md#rights.poll_fd_readwrite).

Rather than annotate each of the call sites, I could instead add the documentation once to the `rights_t` struct in std/os/wasi.zig ?  If so, I'll do the rest of the fields too.

**Comment by squeek502** [comment]
The only intention I had with the inline comments is to note that `POLL_FD_READWRITE` is contingent on the `FD_WRITE`/`FD_READ` rights, i.e. it only grants an extra read-related privilege when `FD_READ` is also set, and it only grants an extra write-related privilege when `FD_WRITE` is set. Just seems like the sort of info that would be useful to have when looking at this bit of code.

Not super important, though, and doc comments might work just as well.

**Comment by andrewrk** [comment]
Thanks.

**Pull request merged**